libraries/redcore/api/hal/hal.php

```diff
@@ -1262,7 +1262,12 @@ public function processPostData($data, $configuration)
 				$fieldAttributes = RApiHalHelper::getXMLElementAttributes($field);
 				$fieldAttributes['transform'] = !is_null($fieldAttributes['transform']) ? $fieldAttributes['transform'] : 'string';
 				$fieldAttributes['defaultValue'] = !is_null($fieldAttributes['defaultValue']) ? $fieldAttributes['defaultValue'] : '';
-				$data[$fieldAttributes['name']] = !is_null($data[$fieldAttributes['name']]) ? $data[$fieldAttributes['name']] : $fieldAttributes['defaultValue'];
+
+				if (!isset($data[$fieldAttributes['name']]) || is_null($data[$fieldAttributes['name']]))
```

Comment:
i don't really see the point of the !is_null, why not just
$fieldAttributes['defaultValue'] = $fieldAttributes['defaultValue'] ?: '';